aioxmpp/ibr/service.py

```diff
@@ -21,27 +21,123 @@
 ########################################################################
 
 import asyncio
-
 import aioxmpp
 import logging
-
+from aioxmpp.service import Service
 from . import xso
 
 
 logger = logging.getLogger(__name__)
 
 
-class RegistrationService:
+@asyncio.coroutine
+def get_registration_fields(xmlstream,
+                            timeout=60,
+                            ):
     """
-    Service implementing XMPP In-Band Registration(:xep:`0077`).
+    A query is sent to the server to obtain the fields that need to be
+    filled to register with the server.
+
+    :param xmlstream: Specifies the stream connected to the server where
+                      the account will be created.
+    :type xmlstream: :class:`aioxmpp.protocol.XMLStream`
+
+    :param timeout: Maximum time in seconds to wait for an IQ response, or
+                    :data:`None` to disable the timeout.
+    :type timeout: :class:`~numbers.Real` or :data:`None`
+
+    :return: :attr:`list`
+    """
+
+    iq = aioxmpp.IQ(
+        to=aioxmpp.JID.fromstr(xmlstream._to),
+        type_=aioxmpp.IQType.GET,
+        payload=xso.Query()
+    )
+    iq.autoset_id()
+
+    reply = yield from aioxmpp.protocol.send_and_wait_for(xmlstream,
+                                                          [iq],
+                                                          [aioxmpp.IQ],
+                                                          timeout=timeout)
+    return reply.payload
+
+
+@asyncio.coroutine
+def register(query_xso,
+             xmlstream,
```

Comment:
This is really only a nitpick, but I prefer the argument order ``xmlstream, query_xso``. This is in line with how other send-type functions operate (first the thing over which data is sent, then the data, and then optional flags).